mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/StrategyDependencyResolver.java

```diff
@@ -29,23 +30,44 @@
     protected static final String SMART_TESTING_STRATEGY_PREFIX = "smart.testing.strategy.";
 
     private final Path propertiesPath; // TODO this could be configurable through system property and with this we need a path
+    private final String[] customStrategies;
 
     StrategyDependencyResolver(Path propertiesPath) {
         this.propertiesPath = propertiesPath;
+        this.customStrategies = null;
     }
 
     StrategyDependencyResolver() {
         this.propertiesPath = null;
+        this.customStrategies = null;
+    }
+
+    StrategyDependencyResolver(String[] customStrategies) {
+        this.propertiesPath = null;
+        this.customStrategies = customStrategies;
     }
 
     Map<String, Dependency> resolveDependencies() {
         final Properties properties = new Properties();
         properties.putAll(loadDefaultMapping());
+        properties.putAll(loadCustomStrategies());
         properties.putAll(loadFromFile());
         properties.putAll(System.getProperties());
```

Comment:
Why do we have this line here? I thought that taking care of getting the values from system property should be a responsibility of the `ObjectMapper` class.

Reply:
btw. having this opportunity of setting custom strategies in place opens a possibility of creating more structured names of the strategies. eg: `smart.testing.strategy.my.cool=org.company:my-cool:version` then this line won't work: https://github.com/arquillian/smart-testing/pull/212/files#diff-22d18b722b6b02c374c2c2b102ed5b67R103

Reply:
nd they are loaded by ObjectMapper, notice that here I am not reading anything from any system property

Reply:
The problem is that then `my.cool` is going to be the name of the strategy. Then for now what it means that when you want to enable it you would need to do something like `mvn .... -Dsmart.testing="affected, my.cool"` which probably it has sense, but just wanted to notice it. I am going to write a test and fix problems.

Reply:
I know, but it should still work. I wouldn't wonder if somebody would use it in this way.